common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -440,12 +441,16 @@ private void processDevicePublish(ChannelHandlerContext ctx, MqttPublishMessage
         try {
             Matcher fwMatcher;
             MqttTransportAdaptor payloadAdaptor = deviceSessionCtx.getPayloadAdaptor();
+            TbMsgMetaData md = null;
+            if (deviceSessionCtx.isMqttTransportType()) {
+                md = createMetadataWithTopic(topicName);
+            }
             if (deviceSessionCtx.isDeviceAttributesTopic(topicName)) {
                 TransportProtos.PostAttributeMsg postAttributeMsg = payloadAdaptor.convertToPostAttributes(deviceSessionCtx, mqttMsg);
-                transportService.process(deviceSessionCtx.getSessionInfo(), postAttributeMsg, getPubAckCallback(ctx, msgId, postAttributeMsg));
+                transportService.process(deviceSessionCtx.getSessionInfo(), postAttributeMsg, md, getPubAckCallback(ctx, msgId, postAttributeMsg));
```

Comment:
I suggest to add a method:
```
getMetadata(deviceSessionCtx, topicName)
```
And use it here instead of variable. The reason for this is that you need this metadata only for attributes and telemetry. You don't use it for other requests.